src/webview/leetCodeSolutionProvider.ts

```diff
@@ -3,24 +3,35 @@
 
 import { ViewColumn } from "vscode";
 import { IProblem } from "../shared";
+import { leetCodePreviewProvider } from "./leetCodePreviewProvider";
 import { ILeetCodeWebviewOption, LeetCodeWebview } from "./LeetCodeWebview";
 import { markdownEngine } from "./markdownEngine";
 
 class LeetCodeSolutionProvider extends LeetCodeWebview {
 
+    protected readonly viewType: string = "leetcode.solution";
     private solution: Solution;
+    private sideMode: boolean = false;
 
-    public async show(solutionString: string, problem: IProblem): Promise<void> {
+    public show(solutionString: string, problem: IProblem): void {
         this.solution = this.parseSolution(solutionString, problem);
+        this.sideMode = leetCodePreviewProvider.isSideMode();
```

Comment:
Please always fetch the setting `enableSideMode` instead of getting it from the preview Provider.

Reply:
This is not the case: solution provider always follows the state of preview provider, rather than whether side mode is enabled.

If `enableSideMode` is true, but preview webview is not in side mode, then solution

Reply:
Ok, if we want to make sure `solution` and `result` are always show in the same column, then I think we should directly call

``` js
if (leetCodePreviewProvider.isSideMode()) {
    ...
}
```

instead of they owning their own `sideMode` field.

Reply:
updated